kpops/components/base_components/models/from_section.py

```diff
@@ -16,26 +16,53 @@ class InputTopicTypes(str, Enum):
 
     INPUT = "input"
     EXTRA = "extra"
+    PATTERN = "pattern"
     INPUT_PATTERN = "input-pattern"
     EXTRA_PATTERN = "extra-pattern"
 
 
 class FromTopic(BaseModel):
     """Input topic
 
-    :param type: Topic type
-    :type type: InputTopicTypes
-    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`
-    :type role: str | None
+    :param type: Topic type, defaults to None
+    :type type: InputTopicTypes | None, optional
+    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`.
+        When `role` is not `None`, `type: pattern` is equal to `type: extra-pattern`, defaults to None
+    :type role: str | None, optional
     """
 
-    type: InputTopicTypes = Field(..., description=describe_attr("type", __doc__))
+    type: InputTopicTypes | None = Field(
+        default=None, description=describe_attr("type", __doc__)
+    )
     role: str | None = Field(default=None, description=describe_attr("role", __doc__))
 
     class Config(DescConfig):
         extra = Extra.forbid
         use_enum_values = True
 
+    def __init__(
+        self, type: InputTopicTypes | None = None, role: str | None = None, **kwargs
+    ) -> None:
+        type = self.__assign_type(type, role)
```

Comment:
would it actually be possible to use a Pydantic `root_validator` instead?

Reply:
Yeah, I added it inside. Should I split it into 2 separate ones?

Reply:
is that possible?

Reply:
it is possible, but now with the breaking changes, I don't think it will be needed